vault/client.go

```diff
@@ -42,10 +42,12 @@ func (c *Client) RenewToken() {
 }
 
 // GetSecret returns the value for a given secret.
-func (c *Client) GetSecret(secretEngine string, path string, keys []string, version int, isBinary bool) (map[string][]byte, error) {
+func (c *Client) GetSecret(secretEngine string, path string, keys []string, version int, isBinary bool, namespace string) (map[string][]byte, error) {
 	// Get the secret for the given path and return the secret data.
 	log.Info(fmt.Sprintf("Read secret %s", path))
 
+	c.client.SetNamespace(namespace)
```

Comment:
```suggestion
    rootVaultNamespace := os.Getenv
    if rootVaultNamespace != "" {
	    c.client.SetNamespace(rootVaultNamespace+namespace)
    } else {
      c.client.SetNamespace(namespace)
    }
```

Reply:
I will change this. I have two more questions:

1. Would it make sense to use `rootVaultNamespace+"/"+namespace`, so that the `vaultNamespace` field in the CR doesn't need a leading `/`?
2. When the `VAULT_NAMESPACE` environment variable isn't set, the authentication will fail, right? So should we return in error in that case instead of just using the `vaultNamespace` field.

Result would look as follows:

```go
rootVaultNamespace := os.Getenv("VAULT_NAMESPACE")
    if rootVaultNamespace != "" {
        c.client.SetNamespace(rootVaultNamespace+"/"+namespace)
    } else {
        return nil, fmt.Errorf("VAULT_NAMESPACE environment variable isn't set")
    }
```

What do you think?

Reply:
Yes, that accomplishes the same goal and provides a better user experience by directing the user to add the env var.

The problem I am seeing is that my initial Auth works but renew token requests are failing.  In my latest comment you will see that GetSecret is having trouble navigating my nested namespace structure.

Reply:
So I adjusted the logic to set the namespace to `rootVaultNamespace+"/"+namespace` and also add a condition to the renew process to that the Vault Namespace header is set to the value from the `VAULT_NAMESPACE` right before the token is renewed.

Reply:
Did you push these to the `dev` tag?

Reply:
Yes